File: queue/src/main/java/br/com/queue/dtos/schedule/statistics/ResponseScheduleDashBoardDto.java
package br.com.queue.dtos.schedule.statistics;

import java.util.List;

public record ResponseScheduleDashBoardDto(

        ResponseCountTotalSchedulesStatisticsDto countTotalScheduleStatistics,
        ResponseSchedulePercentagesStatisticsDto schedulePercentagesStatistics,
        List<ResponseSchedulesCreatedByMonthStatisticsDto> schedulesCreatedByMonth,
        List<ResponseSchedulesCreatedByWeekStatisticsDto> schedulesCreatedByWeek,
        ResponseSchedulesCreatedByDayStatisticsDto scheduleCreatedByDay,
        List<ResponseSchedulesByDepartmentStatisticsDto> schedulesByDepartment,
        List<ResponseSchedulesByServiceStatisticsDto> schedulesByService,
        List<ResponseSchedulesByPriorityStatisticsDto> schedulesByPriority,
        List<ResponseSchedulesByHourStatisticsDto> schedulesByHour

) {
}

File: queue-frontend/src/app/dtos/schedule/statistics/ResponseScheduleDashBoardDto.ts
import { ResponseCountTotalSchedulesStatisticsDto } from "./ResponseCountTotalSchedulesStatisticsDto";
import { ResponseSchedulePercentagesStatisticsDto } from "./ResponseSchedulePercentagesStatisticsDto";
import { ResponseSchedulesByDepartmentStatisticsDto } from "./ResponseSchedulesByDepartmentStatisticsDto";
import { ResponseSchedulesByHourStatisticsDto } from "./ResponseSchedulesByHourStatisticsDto";
import { ResponseSchedulesByPriorityStatisticsDto } from "./ResponseSchedulesByPriorityStatisticsDto";
import { ResponseSchedulesByServiceStatisticsDto } from "./ResponseSchedulesByServiceStatisticsDto";
import { ResponseSchedulesCreatedByDayStatisticsDto } from "./ResponseSchedulesCreatedByDayStatisticsDto";
import { ResponseSchedulesCreatedByMonthStatisticsDto } from "./ResponseSchedulesCreatedByMonthStatisticsDto";
import { ResponseSchedulesCreatedByWeekStatisticsDto } from "./ResponseSchedulesCreatedByWeekStatisticsDto";

export interface ResponseScheduleDashBoardDto {

  countTotalScheduleStatistics: ResponseCountTotalSchedulesStatisticsDto;
  schedulePercentagesStatistics: ResponseSchedulePercentagesStatisticsDto;
  schedulesCreatedByMonth: ResponseSchedulesCreatedByMonthStatisticsDto[];
  schedulesCreatedByWeek: ResponseSchedulesCreatedByWeekStatisticsDto[];
  scheduleCreatedByDay: ResponseSchedulesCreatedByDayStatisticsDto;
  schedulesByDepartment: ResponseSchedulesByDepartmentStatisticsDto[];
  schedulesByService: ResponseSchedulesByServiceStatisticsDto[];
  schedulesByPriority: ResponseSchedulesByPriorityStatisticsDto[];
  schedulesByHour: ResponseSchedulesByHourStatisticsDto[];
}
